frontend/bot/src/Graphql.re

```diff
@@ -0,0 +1,108 @@
+%raw
+"global.fetch = require(\"node-fetch\");";
+
+let endpoint = "localhost:8001/graphql";
+[@bs.module]
+external websocketImpl: SubscriptionsTransportWS.websocketImpl =
+  "isomorphic-ws";
+
+let client = {
+  let subscriptionClient =
+    SubscriptionsTransportWS.subscriptionClient(
+      ~url="ws://" ++ endpoint,
+      ~subscriptionClientConfig=
+        SubscriptionsTransportWS.subscriptionClientConfig(),
+      ~websocketImpl,
+    );
+
+  let forwardSubscription = operation =>
+    subscriptionClient##request(operation);
+
+  let subscriptionExchangeOpts =
+    ReasonUrql.Exchanges.subscriptionExchangeOpts(~forwardSubscription);
+
+  let subscriptionExchange =
+    ReasonUrql.Exchanges.subscriptionExchange(subscriptionExchangeOpts);
+
+  ReasonUrql.Client.make(
+    ~url="http://" ++ endpoint,
+    ~exchanges=
+      Array.append(
+        ReasonUrql.Exchanges.defaultExchanges,
+        [|subscriptionExchange|],
+      ),
+    (),
+  );
+};
+
+type response('a) =
+  | NotFound
+  | Data('a)
+  | Error(string);
+
+type err;
+
+[@bs.deriving abstract]
+type responseJs = {
+  data: Js.Nullable.t(Js.Json.t),
+  [@bs.optional]
+  error: err,
+};
+
+let processResponse = (parse, response) => {
+  let data = response->dataGet->Js.Nullable.toOption->Belt.Option.map(parse);
+  let error = response->errorGet;
```

Comment:
I assume you just pasted this from somewhere, but mind switching to `|>` and not using belt

Reply:
(tablecloth I guess?)

Reply:
🙄